### DISCUSSION TITLE
Publish a `vercel.json` JSON schema

### DISCUSSION THREAD

**Discussion by bitjson** [vercel/community#277] [Help] [answered]
**Is your feature request related to a problem? Please describe.**
I've made plenty of mistakes in `now.json` files, and I regularly spend a lot of time searching through documentation to try to figure out available options.

**Describe the solution you'd like**
If Zeit published a JSON schema for `now.json`, I'd get autocomplete and built-in docs in my IDE. 

You can see a live example at [ide.bitauth.com](https://ide.bitauth.com/). (Click "Import or Restore Template" and try hovering over the default keys.) A lot of compatible IDEs and JSON editing software will automatically start using schemas where available. E.g. try opening `test.json`: 
```json
{ "$schema": "https://bitauth.com/schemas/authentication-template-v0.schema.json" }
```
in VSCode, then use Ctrl + Space to trigger suggestions.

**Describe alternatives you've considered**
This is a nice alternative to type-checking `now.json` with a search engine / combing through docs.

**Additional context**
Continued from https://github.com/zeit/docs/issues/1200:

>For what it's worth, it's actually really easy to generate a schema from TypeScript using [ts-json-schema-generator](https://github.com/vega/ts-json-schema-generator). Here's how I'm using it: https://github.com/bitauth/bitcoin-ts/blob/5d2bd4367468e7c9e07c535cb84723c59b592f5f/package.json#L54
>
>I'm generating [this schema](https://github.com/bitauth/bitcoin-ts/blob/5d2bd4367468e7c9e07c535cb84723c59b592f5f/src/lib/auth/templates/bitauth-authentication-template.schema.json) from [this ts file](https://github.com/bitauth/bitcoin-ts/blob/5d2bd4367468e7c9e07c535cb84723c59b592f5f/src/lib/auth/templates/types.ts).  TypeScript's tooling makes it extremely easy to work with and update.

**Comment by Jianrong-Yu** [comment]
There is one [here](https://json.schemastore.org/now.json), for anyone have not found it yet

**Comment by Jianrong-Yu** [comment]
And another one here: https://github.com/ph1p/vercel-vscode-schemes/tree/master/schemas

**Comment by gogocharli** [comment, marked as answer]
Hello there, we have now published a schema for `vercel.json` which can be found here: https://openapi.vercel.sh/vercel.json
